examples/rust/opensk/src/lib.rs

```diff
@@ -64,25 +67,55 @@ struct WasefireEnv {
     // c: PhantomData<C>,
 }
 
-impl AttestationStore for WasefireEnv {
-    fn get(&mut self, id: &Id) -> Result<Option<Attestation>, Error> {
-        todo!()
+impl Persist for WasefireEnv {
+    fn find(&self, key: usize) -> CtapResult<Option<Vec<u8>>> {
+        let Ok(res) = store::find(key) else {
+            // TODO: What is the correct error code to use here? There doesn't
+            // seem to be an error for value not found or something like that.
+            return Err(CTAP1_ERR_INVALID_COMMAND);
```

Comment:
@ia0 Do we need to convert every error code to a corresponding `Ctap2StatusCode`?
And is this the mapping for the error codes: https://github.com/google/wasefire/blob/dev/opensk/crates/error/src/lib.rs#L144-L179

Reply:
In general, only 4 errors are relevant:
- `Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR`: We got into a internal state that shouldn't have been possible.
- `Ctap2StatusCode::CTAP2_ERR_VENDOR_HARDWARE_FAILURE`: Something broke, and storage detected that it was a hardware failure.
- `Ctap2StatusCode::CTAP2_ERR_KEY_STORE_FULL`: We can't do the operation because storage is full or out of lifetime.
- `Ctap2StatusCode::CTAP1_ERR_OTHER`: Something else entirely, but I don't think you need this.

See the `From` in `libraries/opensk/src/ctap/status_code.rs` in OpenSK for how it converts from the Store errors.

In this particular case, we return a `Result` of an `Option`. Key not found is `Ok(None)`. All errors are likely `Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR)` or `Err(tap2StatusCode::CTAP2_ERR_VENDOR_HARDWARE_FAILURE)`, the idea of `.map_err(convert)?` seems reasonable.

Reply:
Thanks! So I would suggest the following mapping:

```rust
// I'm abusing notations because space is u8 and code is u16, but that's the idea.
match (error.space(), error.code()) {
    (User, NotEnough) => CTAP2_ERR_KEY_STORE_FULL,
    (User | Internal, _) => CTAP2_ERR_VENDOR_INTERNAL_ERROR,
    (World, _) => CTAP2_VENDOR_HARDWARE_FAILURE,
    _ => CTAP2_ERR_OTHER,
}
```

This can be improved in the future if needed.

Reply:
Ok, thank you!

I've made the change but i'm converting the u8 & u16 to `Space` and `Code` and then destructuring in the match. Not sure if there is a more idiomatic way to do this.

Reply:
Looks good. That's even better than what I had in mind.